CMakeLists.txt

```diff
@@ -229,18 +230,32 @@ endif()
 configure_file(${Ginkgo_SOURCE_DIR}/cmake/ginkgo.pc.in
     ${Ginkgo_BINARY_DIR}/ginkgo.pc @ONLY)
 
+# WINDOWS NVCC has " inside the string, add escape charater to avoid config problem.
+ginkgo_modify_flags(CMAKE_CUDA_FLAGS)
+ginkgo_modify_flags(CMAKE_CUDA_FLAGS_DEBUG)
+ginkgo_modify_flags(CMAKE_CUDA_FLAGS_RELEASE)
 ginkgo_install()
 
-set(GINKGO_TEST_INSTALL_COMMAND "${Ginkgo_BINARY_DIR}/test_install/test_install")
-if (GINKGO_BUILD_CUDA)
-    set(GINKGO_TEST_INSTALL_COMMAND "${GINKGO_TEST_INSTALL_COMMAND}" && "${Ginkgo_BINARY_DIR}/test_install/test_install_cuda")
+if(MSVC)
+    # Set path/command with $<CONFIG>
+    set(GINKGO_TEST_INSTALL_COMMAND "${Ginkgo_BINARY_DIR}/test_install/$<CONFIG>/test_install")
+    if(GINKGO_BUILD_CUDA)
+        set(GINKGO_TEST_INSTALL_COMMAND "${GINKGO_TEST_INSTALL_COMMAND}" "${Ginkgo_BINARY_DIR}/test_install/$<CONFIG>/test_install_cuda")
+    endif()
+    set(GINKGO_BUILD_TEST_INSTALL_COMMAND COMMAND ${CMAKE_COMMAND} --build ${Ginkgo_BINARY_DIR}/test_install --config $<CONFIG>)
+else()
+    set(GINKGO_TEST_INSTALL_COMMAND "${Ginkgo_BINARY_DIR}/test_install/test_install")
+    if(GINKGO_BUILD_CUDA)
+        set(GINKGO_TEST_INSTALL_COMMAND "${GINKGO_TEST_INSTALL_COMMAND}" "${Ginkgo_BINARY_DIR}/test_install/test_install_cuda")
+    endif()
+    set(GINKGO_BUILD_TEST_INSTALL_COMMAND COMMAND ${CMAKE_COMMAND} --build ${Ginkgo_BINARY_DIR}/test_install)
```

Comment:
Do we have to keep this separate? We could maybe also use the `$<CONFIG>` version for Linux, or is there an issue I'm not aware of?

Reply:
using `--config` is fine in Linux. cmake ignore the `--config` option
`$<CONFIG>` is the same as `${CMAKE_BUILD_TYPE}` even if using `--config`

I am also fine with adding  `--config` and some comment like `--config is ignored in single-configuration generator` if it is clear enough.

Reply:
Great thanks!